runtime/drivers/duckdb/olap.go

```diff
@@ -181,6 +186,344 @@ func (c *connection) EstimateSize() (int64, bool) {
+// InsertTableAsSelect implements drivers.OLAPStore.
+func (c *connection) InsertTableAsSelect(ctx context.Context, name string, byName bool, sql string) error {
+	var insertByNameClause string
+	if byName {
+		insertByNameClause = "BY NAME"
+	} else {
+		insertByNameClause = ""
+	}
+
+	if !c.config.TableAsView {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("INSERT INTO %s %s (%s)", safeSQLName(name), insertByNameClause, sql),
+			Priority:    1,
+			LongRunning: true,
+		})
+	}
+	version, exist, err := c.tableVersion(name)
+	if err != nil {
+		return err
+	}
+	if !exist {
+		return fmt.Errorf("table %q does not exist", name)
+	}
+	return c.Exec(ctx, &drivers.Statement{
+		Query:       fmt.Sprintf("INSERT INTO %s.default %s (%s)", safeSQLName(dbName(name, version)), insertByNameClause, sql),
+		Priority:    1,
+		LongRunning: true,
+	})
+}
+
+// RenameTable implements drivers.OLAPStore.
+func (c *connection) RenameTable(ctx context.Context, oldName, newName string, view bool) error {
+	if oldName == newName {
+		// todo check case sensitivity
+		return fmt.Errorf("old and new name are same")
+	}
+	if view {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("ALTER VIEW %s RENAME TO %s", safeSQLName(oldName), safeSQLName(newName)),
+			Priority:    100,
+			LongRunning: true,
+		})
+	}
+	if !c.config.TableAsView {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("ALTER TABLE %s RENAME TO %s", safeSQLName(oldName), safeSQLName(newName)),
+			Priority:    100,
+			LongRunning: true,
+		})
+	}
+
+	// ignore query cancellations
+	ctx = context.WithValue(context.Background(), connCtxKey{}, connFromContext(ctx))
+	newSrcDir := filepath.Join(filepath.Dir(c.config.DBFilePath), newName)
+	// new name should not exist
+	exist := dirExist(newSrcDir)
+	if exist {
+		return fmt.Errorf("table %q already exist", newName)
+	}
+
+	oldVersion, exist, err := c.tableVersion(oldName)
+	if err != nil {
+		return err
+	}
+	if !exist {
+		return fmt.Errorf("table %q does not exist", oldName)
+	}
+
+	// drop view
+	err = c.Exec(ctx, &drivers.Statement{
+		Query:       fmt.Sprintf("DROP VIEW IF EXISTS %s", safeSQLName(oldName)),
+		Priority:    100,
+		LongRunning: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	// detach db
+	err = c.Exec(ctx, &drivers.Statement{
+		Query:       fmt.Sprintf("DETACH %s", safeSQLName(dbName(oldName, oldVersion))),
+		Priority:    100,
+		LongRunning: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	// rename dir
+	oldSrcDir := filepath.Join(filepath.Dir(c.config.DBFilePath), oldName)
+	err = os.Rename(oldSrcDir, newSrcDir)
+	if err != nil {
+		return err
+	}
+
+	// rename filename
+	newVersion := fmt.Sprint(time.Now().UnixMilli())
+	newDB := dbName(newName, newVersion)
+	newFilePath := filepath.Join(newSrcDir, fmt.Sprintf("%s.db", newVersion))
+	err = os.Rename(filepath.Join(newSrcDir, fmt.Sprintf("%s.db", oldVersion)), newFilePath)
+	if err != nil {
+		return err
+	}
```

Comment:
Take a closer look at the consistent renaming logic in [Notion](https://www.notion.so/rilldata/Ingest-sources-in-a-new-DB-d35afe9eb0484c68a1ba6bcb4d5f1096?pvs=4#35c05c6633074c8c8f5eb02cb6f4da0a).

It needs to move the `.db` file itself in order to keep serving queries while the renaming is happening